tests/functional/test.cli.run.js

```diff
@@ -47,3 +47,44 @@ describe('web-ext run', () => {
          });
        }));
 });
+
+describe('web-ext run', () => {
```

Comment:
Functional tests are very expensive because they take a lot longer than unit tests. Anytime we add a functional test we have to ask, can this be tested just as well in unit test? For this case I think the answer is yes. You can make a test that passes in a stub version of the `run` function and asserts that `--firefox-binary` is passed to the function like `run({firefox: '/the/path/'})`. You can model it after this test: https://github.com/mozilla/web-ext/blob/master/tests/unit/test.program.js#L283-L294